spec/controllers/queue_items_controller_spec.rb

```diff
@@ -3,72 +3,77 @@
 describe QueueItemsController do
   describe "GET index" do
     it "sets @queue_items" do
-      current_user = Fabricate(:user)
-      session[:user_id] = current_user.id
-      queue_item1 = Fabricate(:queue_item, user: current_user, video_id: 1, position: 1 )
-      queue_item2 = Fabricate(:queue_item, user: current_user, video_id: 2, position: 2)
+      alice = Fabricate(:user)
+      set_current_user(alice)
+      queue_item1 = Fabricate(:queue_item, user: alice, video_id: 1, position: 1 )
+      queue_item2 = Fabricate(:queue_item, user: alice, video_id: 2, position: 2)
       get :index
       expect(assigns(:queue_items)).to match_array([queue_item1, queue_item2])
     end
-    it "redirects to sign_in page for unauthenticted users" do
-      get :index
-      expect(response).to redirect_to(sign_in_path)
+    it_behaves_like "require_sign_in" do 
+      let(:action) { get :index }
     end
   end
 
   describe "POST create" do
     context "authenticated users" do
-      let(:current_user) { Fabricate(:user) }
-      before { session[:user_id] = current_user }
-
       it "creates queue_item record" do
+        set_current_user
         video = Fabricate(:video)
         post :create, video_id: video.id
         expect(QueueItem.count).to eq(1)
       end
       it "renders to video show template after saving queue_item record" do
+        set_current_user
         video = Fabricate(:video)
         post :create, video_id: video.id
         expect(response).to redirect_to(video_path(video))
       end
-      it "creates queue item associated with video" do
+      it "creates queue item associated with video" do  
+        set_current_user
         video = Fabricate(:video)
         post :create, video_id: video.id
         expect(QueueItem.first.video).to eq(video)
       end
       it "creates queue item associated with signed in user" do
+        alice = Fabricate(:user)
+        set_current_user(alice)
         video = Fabricate(:video)
-        post :create,  video_id: video.id
-        expect(QueueItem.first.user).to eq(current_user)
+        post :create, video_id: video.id
+        expect(QueueItem.first.user).to eq(alice)
       end
       it "puts video as last one in queue" do
+        alice = Fabricate(:user)
+        set_current_user(alice)
         monk = Fabricate(:video)
         post :create, video_id: monk.id
         south_park = Fabricate(:video)
         post :create, video_id: south_park.id
-        find_south_park_queue_item = QueueItem.where(video_id: south_park.id, user_id: current_user.id).first
+        find_south_park_queue_item = QueueItem.where(video_id: south_park.id, user_id: alice.id).first
```

Comment:
`find_by` is superior choice compared to `where().first`

HOMEWORK: study why `find_by` is better choice and explain.